File: LQAppleLogin/LQAppleLoginOC/LQAppleLogin/ViewController.m
//
//  ViewController.m
//  LQAppleLogin
//
//  Created by 刘启强 on 2019/10/10.
//  Copyright © 2019 Q.ice. All rights reserved.
//

#import "ViewController.h"
//#import <AuthenticationServices/AuthenticationServices.h>
#import "LQAppleLogin.h"
#import <AuthenticationServices/ASAuthorizationAppleIDButton.h>
@interface ViewController ()

@end

@implementation ViewController
- (void) performExistingAccount {
    
    
}
- (void) checkLoginState {
    
    
}
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    
    ASAuthorizationAppleIDButton *button = [ASAuthorizationAppleIDButton buttonWithType:(ASAuthorizationAppleIDButtonTypeSignIn) style:(ASAuthorizationAppleIDButtonStyleBlack)];
    [button addTarget:self action:@selector(buttonAction:) forControlEvents:(UIControlEventTouchUpInside)];
    
    button.frame = CGRectMake(60, 60, 60, 60);
    [self.view addSubview:button];
    
    [LQAppleLogin checkAuthorizationStateWithUser:@"userid" completeHandler:^(BOOL authorized, NSString * _Nonnull msg) {
        NSLog(@"%d--%@", authorized, msg);
    }];
}

- (void) buttonAction:(ASAuthorizationAppleIDButton *)button {
    NSLog(@"ddddd");
    [[LQAppleLogin shared] loginWithCompleteHandler:^(BOOL successed, NSString * _Nullable user, NSString * _Nullable familyName, NSString * _Nullable givenName, NSString * _Nullable email, NSString * _Nullable password, NSData * _Nullable identityToken, NSData * _Nullable authorizationCode, NSError * _Nullable error, NSString * _Nonnull msg) {
        
        NSLog(@"%@--%@--%@", user, familyName, email);
    }];
//    [self performExistingAccount];
    
}

- (void) firstAuthorization {
    
}



@end

File: LQAppleLogin/LQAppleLoginSwift/LQAppleLogin/ViewController.swift
//
//  ViewController.swift
//  LQAppleLogin
//
//  Created by 刘启强 on 2019/11/7.
//  Copyright © 2019 Q.ice. All rights reserved.
//

import UIKit
import AuthenticationServices

class ViewController: UIViewController {

    var login = LQAppleLogin()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        let button = ASAuthorizationAppleIDButton(authorizationButtonType: .signIn, authorizationButtonStyle: .black)
        button.addTarget(self, action: #selector(buttonAction), for: .touchUpInside)
        button.frame = CGRect(x: 60, y: 60, width: 130, height: 60)
        self.view.addSubview(button)
        
        // Do any additional setup after loading the view.
    }


    @objc func buttonAction() {
        
        login.loginWithCompleteHandler { (success, user, familyName, givenName, email, password, identityToken, authorizationCode, error, msg) in
            print(msg)
        }
    }
}

